docs/batch jobs/configuration/batch-job-configuration.md

```diff
@@ -0,0 +1,62 @@
+# Goal
+This document aims to weigh in on different options for configuring the integration sub-system batch jobs. Examples of job configuration include the location of the source documents used, whom to notify about job outcome, and where to write the produced information.
+
+Options that would require the application to be compiled in order to change the value associated with a configuration were left out. The application configurations should be provided at launch-time.
+
+In the remainder of the document different alternatives that meet the criteria are discussed, and finally the research conclusion is stated.
+
+# Alternatives
+
+## Command Line switches
+
+Each configuration could be passed with the `--` switch on application launch. For example, if we wanted to associate a key configuration with the value `X` we could do so by launch the app with:
+
+`java -jar i-on-integration.jar --key.config=X`
+
+The value `X` would be available at runtime using the annotation @Value("${key.config}").
+
+- Despite being very simple, this approach is error-prone, as each new configuration would have to be added by hand in the command that launches I-On Integration.
+- It would lead to a large and difficult to read java command.
+- Also, different jobs could not have the same key.
+  
+## Spring configuration application.properties
+
+Apart from the default [properties](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html#mail-properties) available at runtime in a spring application, we can define our own inside the application.properties file. This file should be present in a /config subdirectory of the current directory or in the current directory, in the classpath root or in the classpath /config package.
+
+- The main issue with this option is that the configuration file will grow in size as new batch jobs are added. In the long run, that can cause bloating.
+- Launching different instances of the same job would require the properties file to be updated or having a folder per job instance with an application.properties file. To specify the location of the properties file -Dspring.config.location=/path/to/application.properties should be used. Example:
+`java -jar configurations-0.0.1-SNAPSHOT.jar --spring.config.location=./confs/job1/application.properties`
+
+## Additional .properties/.yml files passed in --spring.config.location or spring.config.additional-location
+
+What is done for different instances of the same job (having different application.properties files in different folders) can be done to segregate properties across jobs. In practice, we could define a properties file per job instance.
```

Comment:
- Where are these per-job config files located?
- How is the mapping from job to file made?
- Can we still "project" this configurations into "typed" class instances?